src/compiler/parser.ts

```diff
@@ -4400,8 +4481,8 @@ namespace ts {
         }
 
         function parseArgumentOrArrayLiteralElement(): Expression {
-            return token() === SyntaxKind.DotDotDotToken ? parseSpreadElement() :
-                token() === SyntaxKind.CommaToken ? <Expression>createNode(SyntaxKind.OmittedExpression) :
+            return token() === SyntaxKind.DotDotDotToken ? doInExpressionContext(parseSpreadElement) :

```

Comment:
If this is an argument, shouldn't it *not* be an expression?

Reply:
And you would be justified in thinking that! Because that's certainly true from an AST point of view! However our prior implementation of `isPartOfExpression` always considered these elements as expressions - regardless of where they appeared. It may very well be more correct to actually make these contextual as expected!